earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/filewalker_unittest.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Unit Tests for walking globe files.
+
+#include <algorithm>
+#include <fstream>
+#include <limits.h>
+#include <map>
+#include <string>
+#include <utility>
+#include <gtest/gtest.h>
+#include "common/khFileUtils.h"
+
+#include "fusion/portableglobe/servers/fileunpacker/shared/glc_unpacker.h"
+#include "fusion/portableglobe/servers/fileunpacker/shared/portable_glc_reader.h"
+
+// Test classes for file walker package.
+class FileWalkerTest : public testing::Test {
+ protected:
+  std::string glb_file;
+  std::string glc_file;
+
+  virtual void SetUp() {
+    glb_file = "fusion/testdata/test_data/test.glb";
+    glc_file = "fusion/testdata/test_data/test.glc";
+  }
+
+  // Helper function to find values in lists
+  template<typename T>
+  bool Contains(std::list<T> & element_list, const T & element)
+  {
+	  auto it = std::find(element_list.begin(), element_list.end(), element);
+	  return it != element_list.end();
+  }
+
+  // Helper function to check if a file exists
+  bool FileExists(const char *filename)
+  {
+    std::ifstream ifile(filename);
+    return (bool)ifile;
```

Comment:
I had to hunt down how this works.  Found it at http://www.cplusplus.com/reference/ios/ios/operator_bool/.  👍